python/cudf_polars/cudf_polars/dsl/utils/naming.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Name generation utilities."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+
+__all__ = ["unique_names"]
+
+
+def unique_names(prefix: str) -> Generator[str, None, None]:
+    """
+    Generate unique names with a given prefix.
+
+    Parameters
+    ----------
+    prefix
+        Prefix to give to names
+
+    Notes
+    -----
+    If creating temporary named expressions for a node, create a
```

Comment:
Thoughts on this taking a schema and doing that `unique_names("_" * max(map(len, schema)))` for the caller?

Reply:
Good idea (I take an iterable of names).